Darktheme color changes

Feichtmeier opened the pull request:
The following changes were made only for the dark theme:
- bg color 
- button color 
- border color 
- button color 
- button active/hover/insensitive colors
- backdrop button active/hover/insensitive colors
- some gnome todo tweaks

@madsrh  please check out the look
@clobrano please double check the implementation. especially in _drawing

Despite the must have changes like super bright button hovers and pressed colors I changed the backdrop with the following idea in mind:
- don't change to much from the active colors
- only soften the border and text color


To do:

- [x]  Checkboxes/Radio btns background
- [x] Color of insensitive text in the backdrop

didrocks commented:
A new test snap version is available using: `snap refresh communitheme --channel=edge/gtk-communitheme-pr537`.
Further available updates will track that pull request then.

Switch back to stable or edge snap with `snap refresh communitheme --stable` or `snap refresh communitheme --edge` once you are done with it!
You may want to download as well `gtk-common-themes` snap from `edge/gtk-communitheme-pr537` channel to test your snaps with those changes.

Replace `refresh` with `install` if you haven't installed the snap yet.

madsrh commented:
Looks good @Feichtmeier  😄 

The border-top / box-shadow looks a bit too dark for my taste
![image](https://user-images.githubusercontent.com/3986894/41261764-99808648-6ddd-11e8-8b17-174a32d88971.png)

Disabled borders almost disappears: (but it actually looks kind of cool)
![image](https://user-images.githubusercontent.com/3986894/41261787-aeb1439a-6ddd-11e8-8105-c8c8dad20a61.png)

You didn't work on this one yet but IMO the light gray switch background is too bright and the disabled switch border should be darker:
![image](https://user-images.githubusercontent.com/3986894/41261856-f66588ea-6ddd-11e8-9d2f-1cd450cf96fe.png)

Feichtmeier commented:
@madsrh 
What's your opinion on this dark switch approach?
![loopaolsdasd](https://user-images.githubusercontent.com/15329494/41302774-a82882a2-6e6b-11e8-9a92-9c8ca3a9837b.gif)

madsrh commented:
@Feichtmeier I don't like it, sorry. Just fine tuning what you had wound be better IMO

Feichtmeier commented:
@madsrh give me some help please. I am a bit confused here. Should the slider be dark or light? And should the background be dark or light?

madsrh commented:
I'm on my phone. I'll post something later 😀

Feichtmeier commented:
Alright, meanwhile I brightened the btns a bit and quiet like it:
![image](https://user-images.githubusercontent.com/15329494/41303140-a5c33204-6e6c-11e8-9a1a-3161f00f7aeb.png)
You like that?

clobrano commented on Communitheme/gtk-3.0/_drawing.scss:
```diff
@@ -238,6 +238,7 @@
     @if $flat == true { $c: darken($c, 10%); }
     $_bg: if(lightness($c)<35%, lighten($c, 25%), lighten($c, 5%));
     background-color: $_bg;
+    @if $variant=='dark' and $c != darken($headerbar_bg_color, 10%) { background-color: lighten($c,1%); }
```
This is because on flat==true, $c is equal to darken($headerbar_bg_color, 10%), right?
What do you think about creating a (e.g.) 
_c_flat_headerbar: darken($hb..., 10%)

and then use the this variable in both cases? It would be clearer that the two values are linked even in the future

Feichtmeier replied:
Great idea. I'll try that

clobrano commented on Communitheme/gtk-3.0/_drawing.scss:
```diff
@@ -256,6 +257,7 @@
     $_bg: if(lightness($c)<35%, lighten($c, 18%), darken($c, 7%));
 
     background-color: $_bg;
+    @if $variant=='dark' and $c != darken($headerbar_bg_color,5%) { background-color: darken($button_bg_color,6.5%); }
```
same as above

madsrh commented:
@Feichtmeier 

I don't like the black border here, it looked better before with gray
![image](https://user-images.githubusercontent.com/3986894/41311363-5aa77bdc-6e84-11e8-8c75-7edaa9930b61.png)

For the switches, this was what I was trying to explain 😄
![switches](https://user-images.githubusercontent.com/3986894/41311412-7cba1d38-6e84-11e8-8233-20f823e9e5b5.gif)

Feichtmeier commented:
@madsrh 
![likethis](https://user-images.githubusercontent.com/15329494/41314145-278e68e8-6e8c-11e8-9ad1-67782b5ab72d.gif)
?

madsrh commented:
Did the know inside the switch turn gray? Except that, yes

didrocks commented:
A new test snap version is available using: `snap refresh communitheme --channel=edge/gtk-communitheme-pr537`.
Further available updates will track that pull request then.

Switch back to stable or edge snap with `snap refresh communitheme --stable` or `snap refresh communitheme --edge` once you are done with it!
You may want to download as well `gtk-common-themes` snap from `edge/gtk-communitheme-pr537` channel to test your snaps with those changes.

Replace `refresh` with `install` if you haven't installed the snap yet.

madsrh commented:
@Feichtmeier This is looking **really** good 👍 

I have one small thing 🤦‍♂️  I promised that I wouldn't bring checkboxes/radios up again, BUT...

![image](https://user-images.githubusercontent.com/3986894/41317187-c5906d9a-6e94-11e8-8799-aede473d5072.png)

Inside the square/circle there's another background color than outside! (_I'm overdoing it to illustrate_)
![image](https://user-images.githubusercontent.com/3986894/41317303-14589a4c-6e95-11e8-95fb-83c61d1bab87.png)

This one I'm less sure about, but when hovered shouldn't the entire line be highlighted ❔ 
![image](https://user-images.githubusercontent.com/3986894/41317516-b61a370a-6e95-11e8-8251-f7d94036fe65.png)

madsrh commented:
Software Center 😢 

We can't make it perfect because the color of the stars are hardcoded and it's the same color as the hover gray:
![image](https://user-images.githubusercontent.com/3986894/41318900-3f7cb7a8-6e9a-11e8-9e60-f69b8fdc7852.png)

The bright border looks odd:
![borders_software](https://user-images.githubusercontent.com/3986894/41318983-8cc88064-6e9a-11e8-8b72-82de4814db45.gif)

Feichtmeier commented:
It's never dark

didrocks commented:
A new test snap version is available using: `snap refresh communitheme --channel=edge/gtk-communitheme-pr537`.
Further available updates will track that pull request then.

Switch back to stable or edge snap with `snap refresh communitheme --stable` or `snap refresh communitheme --edge` once you are done with it!
You may want to download as well `gtk-common-themes` snap from `edge/gtk-communitheme-pr537` channel to test your snaps with those changes.

Replace `refresh` with `install` if you haven't installed the snap yet.

didrocks commented:
A new test snap version is available using: `snap refresh communitheme --channel=edge/gtk-communitheme-pr537`.
Further available updates will track that pull request then.

Switch back to stable or edge snap with `snap refresh communitheme --stable` or `snap refresh communitheme --edge` once you are done with it!
You may want to download as well `gtk-common-themes` snap from `edge/gtk-communitheme-pr537` channel to test your snaps with those changes.

Replace `refresh` with `install` if you haven't installed the snap yet.

didrocks commented:
A new test snap version is available using: `snap refresh communitheme --channel=edge/gtk-communitheme-pr537`.
Further available updates will track that pull request then.

Switch back to stable or edge snap with `snap refresh communitheme --stable` or `snap refresh communitheme --edge` once you are done with it!
You may want to download as well `gtk-common-themes` snap from `edge/gtk-communitheme-pr537` channel to test your snaps with those changes.

Replace `refresh` with `install` if you haven't installed the snap yet.

didrocks commented:
A new test snap version is available using: `snap refresh communitheme --channel=edge/gtk-communitheme-pr537`.
Further available updates will track that pull request then.

Switch back to stable or edge snap with `snap refresh communitheme --stable` or `snap refresh communitheme --edge` once you are done with it!
You may want to download as well `gtk-common-themes` snap from `edge/gtk-communitheme-pr537` channel to test your snaps with those changes.

Replace `refresh` with `install` if you haven't installed the snap yet.

didrocks commented:
A new test snap version is available using: `snap refresh communitheme --channel=edge/gtk-communitheme-pr537`.
Further available updates will track that pull request then.

Switch back to stable or edge snap with `snap refresh communitheme --stable` or `snap refresh communitheme --edge` once you are done with it!
You may want to download as well `gtk-common-themes` snap from `edge/gtk-communitheme-pr537` channel to test your snaps with those changes.

Replace `refresh` with `install` if you haven't installed the snap yet.

didrocks commented:
A new test snap version is available using: `snap refresh communitheme --channel=edge/gtk-communitheme-pr537`.
Further available updates will track that pull request then.

Switch back to stable or edge snap with `snap refresh communitheme --stable` or `snap refresh communitheme --edge` once you are done with it!
You may want to download as well `gtk-common-themes` snap from `edge/gtk-communitheme-pr537` channel to test your snaps with those changes.

Replace `refresh` with `install` if you haven't installed the snap yet.

Feichtmeier commented:
@madsrh @clobrano 
Okay lots of improvements since you last checked (dark only):
- fixed some bugs I've created =D
- changed the control structure for color conditions

- changed the color of the backdrop for all btn types 
- changed menu color and border

madsrh commented:
Insensitive text in backdrop disappears
![tdxt](https://user-images.githubusercontent.com/3986894/41380003-7f8d5554-6f18-11e8-8b80-7a2109e8a553.gif)

didrocks commented:
A new test snap version is available using: `snap refresh communitheme --channel=edge/gtk-communitheme-pr537`.
Further available updates will track that pull request then.

Switch back to stable or edge snap with `snap refresh communitheme --stable` or `snap refresh communitheme --edge` once you are done with it!
You may want to download as well `gtk-common-themes` snap from `edge/gtk-communitheme-pr537` channel to test your snaps with those changes.

Replace `refresh` with `install` if you haven't installed the snap yet.

didrocks commented:
A new test snap version is available using: `snap refresh communitheme --channel=edge/gtk-communitheme-pr537`.
Further available updates will track that pull request then.

Switch back to stable or edge snap with `snap refresh communitheme --stable` or `snap refresh communitheme --edge` once you are done with it!
You may want to download as well `gtk-common-themes` snap from `edge/gtk-communitheme-pr537` channel to test your snaps with those changes.

Replace `refresh` with `install` if you haven't installed the snap yet.

didrocks commented:
A new test snap version is available using: `snap refresh communitheme --channel=edge/gtk-communitheme-pr537`.
Further available updates will track that pull request then.

Switch back to stable or edge snap with `snap refresh communitheme --stable` or `snap refresh communitheme --edge` once you are done with it!
You may want to download as well `gtk-common-themes` snap from `edge/gtk-communitheme-pr537` channel to test your snaps with those changes.

Replace `refresh` with `install` if you haven't installed the snap yet.

Feichtmeier commented:
@madsrh 
- fixed radios
- fixed backdrop insensitive labels
- fixed menus border vanishing
Could you check again the menus? 
![image](https://user-images.githubusercontent.com/15329494/41429768-5e56fab0-700e-11e8-8f7c-df3f4f9794b3.png)

Then we could merge it if @clobrano is OK with the code

didrocks commented:
A new test snap version is available using: `snap refresh communitheme --channel=edge/gtk-communitheme-pr537`.
Further available updates will track that pull request then.

Switch back to stable or edge snap with `snap refresh communitheme --stable` or `snap refresh communitheme --edge` once you are done with it!
You may want to download as well `gtk-common-themes` snap from `edge/gtk-communitheme-pr537` channel to test your snaps with those changes.

Replace `refresh` with `install` if you haven't installed the snap yet.

Feichtmeier commented:
I think it's looking quiet good on "real" dark applications now. Like gnome builder:
![screenshot from 2018-06-14 20-10-03](https://user-images.githubusercontent.com/15329494/41429993-11c1bacc-700f-11e8-925d-a35fcdaca5bd.png)
Or ToDo
![image](https://user-images.githubusercontent.com/15329494/41429997-14977d54-700f-11e8-8dc6-c207bca6021c.png)

Or eye of gnome :)

didrocks commented:
A new test snap version is available using: `snap refresh communitheme --channel=edge/gtk-communitheme-pr537`.
Further available updates will track that pull request then.

Switch back to stable or edge snap with `snap refresh communitheme --stable` or `snap refresh communitheme --edge` once you are done with it!
You may want to download as well `gtk-common-themes` snap from `edge/gtk-communitheme-pr537` channel to test your snaps with those changes.

Replace `refresh` with `install` if you haven't installed the snap yet.

madsrh commented:
Really nice work @Feichtmeier 👍 🎉 

The only things I can think of is:
![image](https://user-images.githubusercontent.com/3986894/41435174-480f1688-701e-11e8-8e44-7d21fd90fa67.png)

![image](https://user-images.githubusercontent.com/3986894/41435203-5c79ba88-701e-11e8-991b-ac871dd828da.png)

Feichtmeier commented:
That icon view is on my list for next pr and the text will change anyways soon :)

Feichtmeier commented:
I can push that iconview but it's kinda the wrong pr
